power_grid_model_c/power_grid_model/include/power_grid_model/main_model_impl.hpp

```diff
@@ -804,7 +804,10 @@ class MainModelImpl<ExtraRetrievableTypes<ExtraRetrievableType...>, ComponentLis
     void check_no_experimental_features_used(Options const& options) const {
         if (options.calculation_type == CalculationType::state_estimation &&
             state_.components.template size<GenericCurrentSensor>() > 0) {
-            throw ExperimentalFeature{"State estimation", "current sensors"};
+            if (options.calculation_method == CalculationMethod::newton_raphson) {
+                throw ExperimentalFeature{"Newton-Raphson state estimation is not implemented for current sensors"};
+            }
+            throw ExperimentalFeature{"State estimation with current sensors is experimental"};
```

Comment:
i think this gives the suggestion that we're never going to implement NRSE current sensors.

```suggestion
            throw ExperimentalFeature{"State estimation", "current sensors"};
```

I would just apply the experimental feature change in a follow-up PR

Reply:
The original error message does not differentiate NR from IL and the message alone would suggest NRSE with current sensor is also already experimental like IL. I think it's worth some extra messaging to get the confusion out of the way.

Reply:
hmm... good point. The main issue i have with "not implemented" is that it would need to change it back when we actually start implementing it. It's alright with me since it would only be a short while. In fact, we may even change it back as soon as the release of ILSE current sensor is done. Would that be alright with you?

There is an exception created by sudo-ac that could potentially resolve this ambiguity but it is not correctly working from the python side anymore (i haven't fixed it yet) and it is definitely out of scope of this PR/release to revive that.

Reply:
Cfr. off-line discussion: we'll update the message when relevant